### PR TITLE
[Feature](Routine Load)Support Partial Update

### PR DESCRIPTION
**Pull request by CalvinKirs**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by CalvinKirs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/AlterRoutineLoadStmt.java [hunk 1]
```diff
@@ -111,12 +120,29 @@ public void analyze(Analyzer analyzer) throws UserException {
...
+            return;
+        }
+        RoutineLoadJob job = Env.getCurrentEnv().getRoutineLoadManager()
+                .getJob(getDbName(), getLabel());
+        if (job.isMultiTable()) {
+            throw new AnalysisException("load by PARTIAL_COLUMNS is not supported in multi-table load.");
+        }
+        Database db = Env.getCurrentInternalCatalog().getDbOrAnalysisException(job.getDbFullName());
+        Table table = db.getTableOrAnalysisException(job.getTableName());
+        if (isPartialUpdate && ((OlapTable) table).getKeysType() != KeysType.UNIQUE_KEYS) {
```
Comment (zhannngchen) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/AlterRoutineLoadStmt.java:141]: should use `!((OlapTable) table).getEnableUniqueKeyMergeOnWrite()`

File: fe/fe-core/src/main/java/org/apache/doris/analysis/AlterRoutineLoadStmt.java [hunk 2]
```diff
@@ -111,12 +120,29 @@ public void analyze(Analyzer analyzer) throws UserException {
...
+            throw new AnalysisException("load by PARTIAL_COLUMNS is only supported in unique tables.");
```
Comment (zhannngchen) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/AlterRoutineLoadStmt.java:142]: "in unique table MoW"

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateRoutineLoadStmt.java [hunk 3]
```diff
@@ -339,6 +353,9 @@ public void checkDBTable(Analyzer analyzer) throws AnalysisException {
                 && !(table.getType() == Table.TableType.OLAP && ((OlapTable) table).hasDeleteSign())) {
             throw new AnalysisException("load by MERGE or DELETE need to upgrade table to support batch delete.");
         }
+        if (isPartialUpdate && ((OlapTable) table).getKeysType() != KeysType.UNIQUE_KEYS) {
```
Comment (zhannngchen) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateRoutineLoadStmt.java:356]: ditto

Comment (CalvinKirs) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/CreateRoutineLoadStmt.java]: thx, PTAL again:)

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by CalvinKirs** [comment]
run buildall

**Review by bobhan1 — APPROVED** [review]

**Comment by CalvinKirs** [comment]
run clickbench-new (clickbench)

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.82 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162228900 Bytes

**Comment by CalvinKirs** [comment]
run P0

**Pull request merged**